iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -482,70 +551,48 @@ TEST_F(PortManager_test, AcquiringOneMoreThanMaximumNumberOfNodesFails)
     EXPECT_THAT(nodeData.get_error(), Eq(PortPoolError::NODE_DATA_LIST_FULL));
 }
 
-TEST_F(PortManager_test, DeletingNodeWorks)
+TEST_F(PortManager_test, DeleteNodePortfromMaximumNumberandAddOneIsSuccessful)
 {
     std::string process = "Process";
     std::string node = "Node";
 
     // first acquire all possible NodeData
-    for (unsigned int i = 0U; i < iox::MAX_NODE_NUMBER; i++)
-    {
-        iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
-        iox::NodeName_t newNodeName(iox::cxx::TruncateToCapacity, node + std::to_string(i));
-        auto nodeData = m_portManager->acquireNodeData(newProcessName, newNodeName);
-        ASSERT_THAT(nodeData.has_error(), Eq(false));
-    }
-
-    // test if overflow errors get hit
-    auto errorHandlerCalled{false};
-    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
-        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
-            errorHandlerCalled = true;
-        });
-
-    auto nodeData = m_portManager->acquireNodeData("AnotherProcess", "AnotherNode");
-    ASSERT_THAT(nodeData.has_error(), Eq(true));
-    ASSERT_THAT(errorHandlerCalled, Eq(true));
-    ASSERT_THAT(nodeData.get_error(), Eq(PortPoolError::NODE_DATA_LIST_FULL));
+    exhaustNodePorts(node, process);
 
     // delete one and add one NodeData should be possible now
     unsigned int i = 0U;
     iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
     iox::NodeName_t newNodeName(iox::cxx::TruncateToCapacity, node + std::to_string(i));
+    // this is done because there is no removeNodeData method in the PortManager class
     m_portManager->deletePortsOfProcess(newProcessName);
 
-    nodeData = m_portManager->acquireNodeData(newProcessName, newNodeName);
+    auto nodeData = m_portManager->acquireNodeData(newProcessName, newNodeName);
     EXPECT_THAT(nodeData.has_error(), Eq(false));
```

Comment:
this must be an `ASSERT_THAT` to abort the test in case there is no value to access